edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Http/RegistryApiClient.cs

```diff
@@ -0,0 +1,132 @@
+            return response;
+        }
+
+        public async Task<HttpResponseMessage> GetModuleOnBehalfOfAsync(string actorEdgeDeviceId, GetModuleOnBehalfOfData requestData)
+        {
+            var requesteUri = new Uri(this.baseUri, string.Format(CultureInfo.InvariantCulture, GetModuleOnBehalfOfUriTemplate, actorEdgeDeviceId, ApiVersion));
+            var response = await this.SendRequestAsync(requesteUri, HttpMethod.Post, requestData);
+            return response;
+        }
+
+        public async Task<HttpResponseMessage> ListModulesOnBehalfOfAsync(string actorEdgeDeviceId, ListModulesOnBehalfOfData requestData)
+        {
+            var requesteUri = new Uri(this.baseUri, string.Format(CultureInfo.InvariantCulture, ListModulesOnBehalfOfUriTemplate, actorEdgeDeviceId, ApiVersion));
+            var response = await this.SendRequestAsync(requesteUri, HttpMethod.Post, requestData);
+            return response;
+        }
+
+        public async Task<HttpResponseMessage> DeleteModuleOnBehalfOfAsync(string actorEdgeDeviceId, DeleteModuleOnBehalfOfData requestData)
+        {
+            var requesteUri = new Uri(this.baseUri, string.Format(CultureInfo.InvariantCulture, DeleteModuleOnBehalfOfUriTemplate, actorEdgeDeviceId, ApiVersion));
+            var response = await this.SendRequestAsync(requesteUri, HttpMethod.Post, requestData);
+            return response;
+        }
+
+        async Task<HttpResponseMessage> SendRequestAsync<T>(Uri uri, HttpMethod method, T payload, string etag = null)
+        {
+            HttpClient client = this.proxy
+                .Map(p => new HttpClient(new HttpClientHandler { Proxy = p }, disposeHandler: true))
+                .GetOrElse(() => new HttpClient());
+
+            using (var request = new HttpRequestMessage(method, uri))
+            {
+                string token = await this.tokenProvider.GetTokenAsync(Option.None<TimeSpan>());
+                request.Headers.Add(HttpRequestHeader.Authorization.ToString(), token);
+
+                if (!string.IsNullOrEmpty(etag))
+                {
+                    request.Headers.Add("If-Match", etag);
+                }
+
+                var jsonPayload = JsonConvert.SerializeObject(payload);
+                LogDebug($"Uri={method} {request.RequestUri}");
+                LogDebug($"Token={token}");
+                LogDebug($"Content={jsonPayload}");
+                request.Content = new StringContent(jsonPayload, Encoding.UTF8, MediaType);
+                HttpResponseMessage response = await client.SendAsync(request);
+                return response;
+            }
+        }
+
+        static readonly ILogger Log = Logger.Factory.CreateLogger<RegistryApiClient>();
+
+        public static void LogDebug(string message)
+        {
+            Log.LogDebug(9999, message);
+        }
+    }
+
+    public class CreateOrUpdateModuleOnBehalfOfData
```

Comment:
- Does this class have to be public? If not make it private and nested
- Pls make this immutable

Reply:
Yes, as this class requires to be public.